source/Meadow-Endpoints.js

```diff
@@ -435,8 +435,14 @@ var MeadowEndpoints = function()
 				return tmpCallback('Endpoint \'' + pMethod + '\' does not exist!'); //might be better as an exception
 			}
 
-			//TODO: should switch depending on type
+			// TODO: should switch depending on type
+			// TODO: should we keep this around, just make a deep copy of 'pOptions'
 			var pRequest = {params: pData, formattedParams: pData, body: pData};
+			if (typeof(pOptions) === 'object') {
+				// copy header (function) and headers (raw headers)
+				pRequest.header = pOptions.header;
+				pRequest.headers = pOptions.headers;
```

Comment:
```suggestion
				pRequest.header = pOptions.header.bind(pOptions);
```
We could this to abstract way the internal workings.

Reply:
I wonder if there are cases where there are multiple levels of "invoke recursion" where a "fake" endpoint would be passed into those. Do we need to make sure that will work?